M2/Macaulay2/m2/exports.m2

```diff
@@ -1247,6 +1247,8 @@ exportMutable {
 	"recursionLimit",
 	"scriptCommandLine",
 	"stopIfError",
+	"texMathStart",
```

Comment:
Why do these have to be mutable?

Reply:
you mean `texMathStart` and `texMathEnd`? the idea is that in different circumstances you may want different delimiters. The direct application here is that webapp.m2 redefines them for its own purposes, but we still want, when the user types
```
tex(ZZ[x,y,z])
```
the output to have the usual delimiters `$`.

Reply:
But since you introduce
```
tex Thing := x -> concatenate(texMathStart,texMath x,texMathEnd)
```
, if the user types
```
tex(ZZ[x,y,z])
```
, he will get your delimiters, not the usual ones.

What is the purpose your web app has in mind in trying to redefine that?

Reply:
No, it won't. texMathStart and texMathEnd are only redefined *during the output process*. 
To make sure, I just tried it in the web app, and sure enough
```
i1 : tex(ZZ[x,y,z])
o1 =  ${\mathbb Z}\left[x,y,z\right]$
```

Reply:
What if the user is doing things in various threads?  Then they might be doing something else on another thread during the output process.

Of course, much else in the system is not thread-safe, but every little bit of thread-safety can help eventually.

Reply:
OK that's problem. Is there any canonical solution to this kind of thread-safety issues?
Otherwise I can revert to how it was before: I had my own routine `modifiedTex` (or whatever) which did exactly the same as `tex` except it changed the delimiters. I find this solution much less satisfactory.

Reply:
I'm just guessing, but perhaps your webapp top level loop is formatting all the output expressions in tex for display.  So you take an output expression `e` and try to format it as math.  You can use `texMath e` to get the TeX code for it, and then use something else to display it.  Why bother with `tex e` and its behavior at all?

Reply:
that's right. that's how it used to be; but I still need to wrap the tex to let the web app know that the output is tex (as opposed to plain html); that's where the `modifiedTex` (actually I think it was called `texWrap` but anyway) came in. I thought I was being clever merging it with the ordinary `tex` method, but this thread-safety issue can't be avoided, then I suppose I'll have to revert.

Reply:
removed the whole tex redefinition. this PR is getting smaller and smaller!